python/uv/__main__.py

```diff
@@ -9,35 +9,34 @@ def _detect_virtualenv() -> str:
     Find the virtual environment path for the current Python executable.
     """
 
-    # If it's already set, then just use it
-    value = os.getenv("VIRTUAL_ENV")
-    if value:
+    # If it's already set, then use it
+    if value := os.getenv("VIRTUAL_ENV"):
         return value
 
     # Otherwise, check if we're in a venv
     venv_marker = os.path.join(sys.prefix, "pyvenv.cfg")
-
     if os.path.exists(venv_marker):
         return sys.prefix
 
     return ""
 
+
 def _run() -> None:
     uv = os.fsdecode(find_uv_bin())
 
     env = os.environ.copy()
-    venv = _detect_virtualenv()
-    if venv:
+    if venv := _detect_virtualenv():
         env.setdefault("VIRTUAL_ENV", venv)
 
+    cmd = [uv, *sys.argv[1:]]
+
     if sys.platform == "win32":
         import subprocess
 
-        completed_process = subprocess.run([uv, *sys.argv[1:]], env=env)
-        sys.exit(completed_process.returncode)
-    else:
-        os.execvpe(uv, [uv, *sys.argv[1:]], env=env)
+        completed_process = subprocess.run(cmd, env=env)
+        raise SystemExit(completed_process.returncode)
 
+    os.execvpe(uv, cmd, env=env)
```

Comment:
Hmm, I kinda prefer the `if`/`else` here -- I feel like it makes it more explicit that it's a binary choice: for Windows we do one thing; for all other platforms, we do a different thing

Reply:
But raise SystemExit makes it redundant, 🤔 so I prefer this 😆 the raise makes it explicit that the execution of that branch ends there.

Reply:
I agree with you that calling `sys.exit` is redundant, since all it does internally is raise `SystemExit`, so I too prefer raising `SystemExit` directly as a matter of style. But I'd still personally find it more readable to do this:

```py
    if sys.platform == "win32":
        import subprocess

        completed_process = subprocess.run(cmd, env=env)
        raise SystemExit(completed_process.returncode)
    else:
        os.execvpe(uv, cmd, env=env)
```

even though, as you say, the `else` is, strictly speaking, redundant.

Make a bikesheddy PR, expect bikesheddy review comments ;)

Reply:
I prefer my style, but if there's one more maintainer vote on the previous form can revert 👍